From c370d82495b4634c0796e6fd9f291049a3ced391 Mon Sep 17 00:00:00 2001
From: Giuseppe CAVALLARO <peppe.cavallaro@st.com>
Date: Wed, 27 May 2009 16:17:30 +0200
Subject: [PATCH] stm: fixed the STBus Bridge configuration while using the gmac0 on the stx7141c2.

Only the GMAC1 was fixed for the stx7141 cut 2, previously.
This patch extends the same fix (see comment within the code) for the
GMAC0.

Signed-off-by: Giuseppe Cavallaro <peppe.cavallaro@st.com>
Signed-off-by: Stuart Menefy <stuart.menefy@st.com>
---
 arch/sh/kernel/cpu/sh4/setup-stx7141.c |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/sh/kernel/cpu/sh4/setup-stx7141.c b/arch/sh/kernel/cpu/sh4/setup-stx7141.c
index cdf7850..0279421 100644
--- a/arch/sh/kernel/cpu/sh4/setup-stx7141.c
+++ b/arch/sh/kernel/cpu/sh4/setup-stx7141.c
@@ -550,7 +550,6 @@ void __init stx7141_configure_pata(int bank, int pc_mode, int irq)
 }
 
 /* Ethernet MAC resources -------------------------------------------------- */
-#define AHB_STBUS_BASE   0xFD118000
 #define AD_CONFIG_OFFSET 0x7000
 #define READ_AHEAD_MASK  0xFFCFFFFF
 
@@ -665,10 +664,11 @@ void stx7141_configure_ethernet(int port, int reverse_mii, int mode,
 
 	/* Cut 2 of 7141 has AHB wrapper bug for ethernet gmac */
 	/* Need to disable read-ahead - performance impact     */
-	if (cpu_data->cut_major == 2)
-		writel(readl(AHB_STBUS_BASE+AD_CONFIG_OFFSET) & READ_AHEAD_MASK,
-				AHB_STBUS_BASE + AD_CONFIG_OFFSET);
-
+	if (cpu_data->cut_major == 2) {
+		unsigned long base = stx7141eth_devices[port].resource->start;
+		writel(readl(base + AD_CONFIG_OFFSET) & READ_AHEAD_MASK,
+				base + AD_CONFIG_OFFSET);
+	}
 	/* gmac_en: GMAC Enable */
 	sc = sysconf_claim(SYS_CFG, 7, 16+port, 16+port, "stmmac");
 	sysconf_write(sc, 1);
-- 
1.6.0.6

